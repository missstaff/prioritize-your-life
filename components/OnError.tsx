import { router } from "expo-router";
import { ScaledSheet } from "react-native-size-matters";
import AppThemedText from "./app_components/AppThemedText";
import AppThemedView from "./app_components/AppThemedView";

const OnError = (error: any) => {
  return (
    <AppThemedView style={{}}>
      <AppThemedText>Error: {error.message}</AppThemedText>
      <AppThemedText
        style={styles.text}
        type="link"
        onPress={() => router.push("/")}
      >
        Home
      </AppThemedText>
    </AppThemedView>
  );
};

const styles = ScaledSheet.create({
  container: {
    alignItems: "center",
    display: "flex",
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
  },
  text: {
    fontSize: "16@s",
    textAlign: "center",
  },
});

export default OnError;
