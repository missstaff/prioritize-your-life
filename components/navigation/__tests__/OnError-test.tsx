import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";
import OnError from "../../OnError";

jest.mock("expo-router", () => ({
  router: {
    push: jest.fn(),
  },
}));

describe("OnError", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("matches snapshot", () => {
    const { toJSON } = render(
      <OnError error={{ message: "Test error message" }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders the error message and Home link correctly", () => {
    const mockError = { message: "Test error message" };

    const { getByText } = render(<OnError error={mockError} />);

    //     // Check if the error message is rendered
    expect(getByText("Error: ")).toBeTruthy();

    //     // Check if the Home link is rendered
    const homeLink = getByText("Home");
    expect(homeLink).toBeTruthy();
  });

  it("navigates to the home page when the Home link is pressed", () => {
    const mockError = { message: "Test error message" };

    const { getByText } = render(<OnError error={mockError} />);

    fireEvent.press(getByText("Home"));

    expect(router.push).toHaveBeenCalledWith("/");
  });
});
