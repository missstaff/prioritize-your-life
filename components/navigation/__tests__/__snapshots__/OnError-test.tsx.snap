// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnError matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      {},
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ECEDEE",
        },
        {
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Error: 
  </Text>
  <Text
    onPress={[Function]}
    style={
      [
        {
          "color": "#ECEDEE",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "color": "#CD03EF",
          "fontSize": 34.285714285714285,
          "lineHeight": 64.28571428571428,
        },
        {
          "fontSize": 34.285714285714285,
          "textAlign": "center",
        },
      ]
    }
  >
    Home
  </Text>
</View>
`;
